provider/aws.go

```diff
@@ -136,19 +136,15 @@ func (p *AWSProvider) Records() (endpoints []*endpoint.Endpoint, _ error) {
 		for _, r := range resp.ResourceRecordSets {
 			// TODO(linki, ownership): Remove once ownership system is in place.
 			// See: https://github.com/kubernetes-incubator/external-dns/pull/122/files/74e2c3d3e237411e619aefc5aab694742001cdec#r109863370
-			switch aws.StringValue(r.Type) {
-			case route53.RRTypeA, route53.RRTypeCname, route53.RRTypeTxt:
-				break
-			default:
-				continue
-			}
 
-			for _, rr := range r.ResourceRecords {
-				endpoints = append(endpoints, endpoint.NewEndpoint(aws.StringValue(r.Name), aws.StringValue(rr.Value), aws.StringValue(r.Type)))
-			}
+			if ok := recordTypeFilter(aws.StringValue(r.Type)); ok != false {
```

Comment:
It would be nicer to reverse the if statement to:
```go
if !recordTypeFilter(aws.StringValue(r.Type)) {
  continue
}
```

* the coverage report can show that both cases are covered
* keep intendation of original code results in a better diff
* no need for `ok` variable